core/schema.rs

```diff
@@ -1318,7 +1318,8 @@ impl From<&ColumnDefinition> for Column {
 ///
 /// Note that the order of the rules for determining column affinity is important. A column whose declared type is "CHARINT" will match both rules 1 and 2 but the first rule takes precedence and so the column affinity will be INTEGER.
 pub fn affinity(datatype: &str) -> Affinity {
-    // Note: callers of this function must ensure that the datatype is uppercase.
```

Comment:
although I failed to mention it in the code comment, I think the reason was that it made some queries significantly slower. do we have to make this change?